src/app/clusters/bindings/BindingManager.h

```diff
@@ -0,0 +1,126 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/CASESessionManager.h>
+#include <app/server/Server.h>
+#include <app/util/binding-table.h>
+
+namespace chip {
+
+using BoundDeviceChangedHandler = void (*)(EndpointId localEndpoint, EndpointId remoteEndpoint, ClusterId clusterId,
+                                           OperationalDeviceProxy * peer_device);
+
+class BindingManager
```

Comment:
Could you add some documentation about purpose and usage? 

Generally something that is not re-stating the class name: I know this is a  "a singleton manager for cluster bindings" by reading the code, but what are bindings, what does it interact with,   why is there a single change handler, what is an app server and what is it used for, etc.